### ISSUE TITLE
Question: Plug in element(s) from XSD 1 into element of XSD 2 when parsing

### ISSUE THREAD

**Issue by afuetterer** [tefra/xsdata#868]
Hi,

I have another question regarding the usage of this awesome library.

The [XSD](https://www.openarchives.org/OAI/2.0/OAI-PMH.xsd) of the OAI-PMH standard describes a `recordType`. The recordType has an element metadata of type `metadataType`. So far this is straight forward and I set up these classes with xsdata alright.

But the `metadataType` actually uses another XSD to describe the record.
> Metadata must be expressed in XML that complies with another XML Schema (namespace=#other). Metadata must be explicitly qualified in the response.

If I know this other XSD in advance, can I somehow plug this in during the parsing?

TLDR:

- I would like to parse OAI-PMH XML with the classes generated from [OAI-PMH.xsd.](https://www.openarchives.org/OAI/2.0/OAI-PMH.xsd)
- I would then like to somehow parse the metadata with classes from another XSD (e.g. https://www.openarchives.org/OAI/2.0/oai_dc.xsd), this second set of classes should be dynamically changeable

I hope this description is understandable.

Is this possible?

**Comment by tefra** [comment]
It's not possible without modifying the xsd files manually

**Comment by afuetterer** [comment]
Thank you for your suggestion. Actually, the problem goes even further.

The metadata class can be described with different XSDs, which I would like to plug in dynamically (at runtime).

So data from API 1 (using OAI-PMH.xsd) could look like this:
```xml
<OAI-PMH>
...
<Record>
...
<metadata>
**[elements from XSD 2]**
</metadata>
...
</Record>
...
</OAI-PMH>
```

And data from API 2 (using OAI-PMH.xsd) could look like this:

```xml
<OAI-PMH>
...
<Record>
...
<metadata>
**[elements from XSD 3]**
</metadata>
...
</Record>
...
</OAI-PMH>
```

It would be nice to be able to change the `metadata` parsing "on the fly" and to plug in classes from XSD 2 or XSD 3. Is this clearer?

Is this possible with a custom event handler? Should I parse the oai-pmh and metadata seperataly maybe?

**Comment by afuetterer** [comment]
Maybe this is somehow similar to #863?

Can you give an example of a custom xml handler, that skips certain xml element when parsing? Maybe I can use the dataclasses from the other xsds individually as a second step?

Can I keep the original xml while parsing to do second parser step later?

If this is documented, I apologize, but I could not find such a thing in the docs.
